.github/workflows/support-window.yml

```diff
@@ -1,52 +1,49 @@
 name: Update support window diagram
 on:
-  # When our dependencies change
+  # When our dependencies change.
   push:
     paths:
       - .github/workflows/support-window.yml
       - package.json
-      - scripts/support-window.ts
-      - scripts/tsconfig.json
-  # Manually, when TypeScript is released
+      - scripts/support-window.js
+      - scripts/jsconfig.json
+  # Manually, when TypeScript is released.
   # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
   workflow_dispatch:
-permissions:
-  contents: read
-
 jobs:
   support-window:
     permissions:
-      contents: write  # for Git to git push
-    if: github.repository == 'DefinitelyTyped/DefinitelyTyped'
```

Comment:
I have a bad feeling about changing this. I'm imagining a case where a fork merges from upstream, which changes one of the applicable files, which then triggers the CI task to run (wasting CI time). If upstream hasn't yet had the task to update the diagram run (a small window), then the fork's CI run will apply a commit while upstream is applying a commit and leave the fork in a bad state.

If this really needs to run on forks, then I would probably want this check to instead be left at the bottom such that forks won't have these commits auto-applied (maybe fail saying things are out of date?).

Reply:
I'm still apprehensive about letting this run outside of DT just because it'll run when people merge from upstream and the diagram changes. These days, forks by default thankfully do not enable GHA, but I know that some still will and I have seen people's PRs updating via tasks in our GHA config, so it'd be nice to limit that.

At the very least, I think what this should do is skip the push when the repo isn't DT. That way, `master` is always updated, but nobody on other branches gets a surprising commit on their branch that they have to deal with pulling/merging again. Instead, the CI will say "you forgot to run the script". Or, if it fails on their own `master` branch, it won't modify it.

Reply:
I'm not a fan of the "you forgot to run the script" error ... We want to prevent surprise commits to people's branches (PRs) and forks (master), no argument there (I've seen that too). It's desirable to commit the SVG to branches that change it, however, so the diagram is part of the changes/review.

Can we have it both ways? That's what I aimed for in this PR. The only pathological case is when:
1. You change the SVG (now limited to `support-window.js`, etc. edits, and not all of those will change the SVG).
2. Land the script changes without the SVG changes. GHA is there to prevent that by pushing the commit to the branch, but it's not guaranteed.
3. Someone syncs from upstream. Again, GHA will push the commit to master if not the branch, but there's a small (+/- 1 minute) window of opportunity in this case.
4. The synced fork runs GHA. People can disable that and it's off by default these days.
5. If all of those line up, there'll be two identical commits, to the synced fork and to master, with different hashes.
   But Git rebase is smart enough to handle that: The fork's commit will be replaced by the next sync. Git merge is smart enough too, I think?

I don't think there are other risks? So I think this PR sufficiently addresses surprise commits, without needing the "you forgot to run the script" error?